### PR TITLE
Add permute, arange

### PR DESCRIPTION
**Pull request by eellison**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #64323 Add embedding shape analysis
* #64121 Max Pool with indices
* #63530 Add Maxpool to shape analysis / Opinfo
* #64300 Add symbolic shape comparison optimization
* #64299 Refactor to use shape arguments
* #63516 Add view with negative dim
* #63615 Generalize expand logic
* **#63407 Add permute, arange**
* #63365 Add support for slice, selec twith int, index_select
* #63099 Add squeeze, unsqueeze, transpose shape functins
* #63050 Add batch of unary functions

Differential Revision: [D30738149](https://our.internmc.facebook.com/intern/diff/D30738149)

### CODE REVIEW

File: torch/csrc/jit/passes/peephole_non_tensor.cpp [hunk 1]
```diff
@@ -193,13 +194,14 @@ struct PeepholeOptimizeNonTensorImpl {
...
-          GRAPH_UPDATE(
-              "Removing ", getHeader(node), " as input is already an integer");
-          node->output()->replaceAllUsesWith(node->input());
-          changed = true;
-        }
+      } else if (
+          (node->kind() == aten::Int || node->kind() == aten::ceil) &&
+          node->inputs().size() == 1 &&
+          node->input()->type()->cast<IntType>()) {
+        GRAPH_UPDATE(
```
Comment (makslevental) [inline torch/csrc/jit/passes/peephole_non_tensor.cpp:201]: TIL

**Review by makslevental — APPROVED** [review]
LGTM

**Review by desertfire — APPROVED** [review]

**Comment by eellison** [comment]
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30738149).

**Comment by eellison** [comment]
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30738149).

**Comment by eellison** [comment]
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30738149).

**Pull request closed without merging**